lib/studies/reply/adaptive_nav.dart

```diff
@@ -1089,61 +1097,56 @@ class _ReplyFabState extends State<_ReplyFab>
-      openBuilder: (context, closedContainer) {
-        return const ComposePage();
-      },
-      openColor: theme.cardColor,
-      closedShape:
-          isDesktop & widget.extended ? const StadiumBorder() : circleFabBorder,
-      closedColor: theme.colorScheme.secondary,
-      closedElevation: 6,
-      closedBuilder: (context, openContainer) {
-        return Consumer<EmailStore>(
-          builder: (context, model, child) {
-            final onMailView = model.onMailView;
-            final fabSwitcher = _FadeThroughTransitionSwitcher(
-              fillColor: Colors.transparent,
-              child: onMailView
-                  ? Icon(
-                      Icons.reply_all,
-                      key: fabKey,
-                      color: Colors.black,
-                    )
-                  : const Icon(
-                      Icons.create,
-                      color: Colors.black,
-                    ),
-            );
-            final tooltip = onMailView ? 'Reply' : 'Compose';
-
-            if (isDesktop) {
-              return AnimatedSize(
-                vsync: this,
-                curve: Curves.fastOutSlowIn,
-                duration: _kAnimationDuration,
-                child: FloatingActionButton.extended(
-                  heroTag: 'Rail FAB',
-                  tooltip: tooltip,
-                  isExtended: widget.extended,
-                  onPressed: openContainer,
-                  label: Row(
-                    children: [
-                      fabSwitcher,
-                      SizedBox(width: widget.extended ? 16 : 0),
-                      if (widget.extended)
-                        Text(
-                          tooltip.toUpperCase(),
-                          style: Theme.of(context).textTheme.headline5.copyWith(
-                                fontSize: 16,
-                                color: theme.colorScheme.onSecondary,
-                              ),
+    return Consumer<EmailStore>(
+      builder: (context, model, child) {
+        final onMailView = model.onMailView;
+        final fabSwitcher = _FadeThroughTransitionSwitcher(
+          fillColor: Colors.transparent,
+          child: onMailView
+              ? Icon(
+                  Icons.reply_all,
+                  key: fabKey,
+                  color: Colors.black,
+                )
+              : const Icon(
+                  Icons.create,
+                  color: Colors.black,
+                ),
+        );
+        final tooltip = onMailView ? 'Reply' : 'Compose';
+
+        if (isDesktop) {
+          return FloatingActionButton.extended(
+            heroTag: 'Rail FAB',
+            tooltip: tooltip,
+            isExtended: widget.extended,
+            onPressed: () =>
+                desktopMailNavKey.currentState.pushNamed(ReplyApp.composeRoute),
+            label: Row(
+              children: [
+                fabSwitcher,
+                SizedBox(width: widget.extended ? 16 : 0),
```

Comment:
You can move this sizedbox down inside the if, then you don't need to have the condition on its width.
```
if (widget.extended) ..[
  SizedBox(),
  Text(),
]
```

Reply:
Thanks!